src/blog/2024/04/using-mongodb-with-node-red.md

```diff
@@ -0,0 +1,281 @@
+
+!["Screenshot displaying connections of wires in the 'Drop collecton from Database' flow"](./images/using-mongodb-with-node-red-mongodb-drop-flow.png "Screenshot displaying connections of wires in the 'Drop collecton from Database' flow"){data-zoomable}
+
+## Debugging the operations
+
+To debug MongoDB operations in Node-RED, simply add a debug node after the MongoDB4 node. This enables you to monitor the execution of operations and diagnose any errors that may occur. Below are sample debug messages indicating successful operations for each operation we have covered in this guide.
+
+```
+// Message received after insert operation successful
+{
+    "acknowledged": true,
+    "insertedId": "BKoIzMuW" // ID of inserted data
+}
+
+// Message received after update operation successful
+{
+    "acknowledged": true,
+    "modifiedCount": 1, // Indicates that one document is updated
+    "upsertedId": null,
+    "upsertedCount": 0,
+    "matchedCount": 1 // Indicates one document is matched for the provided query
+}
+
+// Message received after delete operation successful
+{
+    "acknowledged": true,
+    "deletedCount": 1 // Indicates that one document is deleted
+}
+
+// To indicate collection is dropped successfully and true value will be returned
+```
+
+### Deploying the flow
+
+!["Screenshot displaying flow of CRM System"](./images/using-mongodb-with-node-red-crm-system-node-red-flow.png "Screenshot displaying flow of CRM System"){data-zoomable}
+
+- With your flow updated to include the above, click the "Deploy" button in the top-right of the Node-RED Editor.
+- Locate the 'Open Dashboard' button at the top-right corner of the Dashboard 2.0 sidebar and click on it to navigate to the dashboard.
+
+Now To drop tables, and retrieve table data, click on the 'Inject Node' button. For customer data insertion, updates, and deletions, navigate to fill in the corresponding forms and submit.
+
+!["Screenshot displaying dashboard view of CRM System"](./images/using-mongodb-with-node-red-crm-system-node-red-dashboard-view.png "Screenshot displaying dashboard view of CRM System"){data-zoomable}
+
+## Additional resources
+
+This section contains helpful links to other content. It isn’t required, so consider it supplemental.
+
+- [MongoDB best practices](https://www.mongodb.com/basics/best-practices) for to increase performance
+- [Ebook: MongoDB in manufacturing](https://www.mongodb.com/collateral/manufacturing-with-mongodb).
+- [Manufacturing & Automotive Industry Knowledge Accelerator](https://learn.mongodb.com/learning-paths/manufacturing-automotive-industry-knowledge-accelerator)
+
+## Conclusion
+
+In this guide, we've covered the integration of MongoDB with Node-RED to create dynamic applications. We've walked through the foundation, setup process, demonstrated how to perform CRUD operations, and discussed debugging techniques. Additionally, to enhance understanding and engagement we've built a Customer Relationship Management system.
```

Comment:
Should we add a link to the blueprint that's available too?

Reply:
Yes, let's include the link to the blueprint.